src/core/lib/slice/slice_internal.h

```diff
@@ -312,4 +312,21 @@ grpc_slice grpc_slice_from_moved_string(grpc_core::UniquePtr<char> p);
 // 0. All other slices will return the size of the allocated chars.
 size_t grpc_slice_memory_usage(grpc_slice s);
 
+void grpc_slice_buffer_transfer(grpc_slice_buffer* src, grpc_slice_buffer* dst);
+void grpc_do_slice_buffer_move_into(grpc_slice_buffer* src,
```

Comment:
In the global C namespace, it's very important that functions are prefixed for the area of code that they belong to.  This name should start with `grpc_slice_buffer_`.

Reply:
Updated; they're now prefixed with _helper to indicate that they're effectively like file-level static methods from the POV of users.

Reply:
Sorry, this is insufficient.  The only case where you can ignore these naming rules is for something defined in a .cc file as `static` or in an anonymous namespace, because those are purely local -- they do not get turned into symbols visible to the linker and therefore cannot conflict with anything defined in any other module.  Functions like this that are defined in a .h file are actually turned into symbols visible to the linker and must follow naming rules to avoid conflicts.

Note that if we converted this to a C++ class, these could be private methods instead of standalone functions, in which case this namespacing issue would go away.